core/types/transaction.go

```diff
@@ -521,6 +521,24 @@ func (t *TxWithMinerFee) SBundle() *SimSBundle {
 	return t.order.AsSBundle()
 }
 
+func (t *TxWithMinerFee) Price() *big.Int {
+	return new(big.Int).Set(t.minerFee)
+}
+
+func (t *TxWithMinerFee) Profit(baseFee *big.Int) *big.Int {
+	if tx := t.Tx(); tx != nil {
+		profit := new(big.Int).Sub(tx.GasPrice(), baseFee)
+		profit.Mul(profit, new(big.Int).SetUint64(tx.Gas()))
```

Comment:
Do you have suggestion for what we could do instead?

Reply:
cc: @Ruteri @avalonche

Reply:
What I can suggest is to re-insert the tx after its apply to the block if its total profit (or the egp) changes enough to move it to the next slot by more than some 1%
Simulating txs up front is also on the table I guess

Reply:
> Simulating txs up front is also on the table

For the scope of this PR, I'd like to avoid simulating the transaction due to the performance hit. 

> Txs will be a bit abusable here.

I agree. I spoke with Bert and we think that this could be mitigated by leveraging the broader builder network and constraining the subset of builders that use this algorithm. Open to other ideas as well @dvush @Ruteri @avalonche

Reply:
It can work because we can be sure that MEV is in the bundles but not txs so txs should not conflict with bundles that much. 

But its also can be wrong because it will prioritize mempool swaps over bundles. Here is univ2 router txs https://etherscan.io/txs?a=0x7a250d5630b4cf539739df2c5dacb4c659f2488d gas limit is much bigger than gas used.

Reply:
I updated to use actual gas used on reinsert